src/Management/src/Endpoint/Middleware/EndpointMiddleware.cs

```diff
@@ -38,24 +38,24 @@ protected EndpointMiddleware(IEndpoint<TResult> endpoint, IOptionsMonitor<Manage
         Endpoint = endpoint;
     }
 
-    public virtual string HandleRequest()
+    public virtual async Task<string> HandleRequestAsync(CancellationToken cancellationToken)
     {
-        TResult result = Endpoint.Invoke();
-        return Serialize(result);
+        TResult result = await Endpoint.InvokeAsync(cancellationToken);
+        return await Task.Run(() => Serialize(result), cancellationToken);
```

Comment:
I am reopening this, as there are many more places inside Management where a separate thread is kicked off instead of wrapping the result.

Reply:
How would you pass the cancellationtoken then?

Reply:
Its not completely pointless because if the task is not already running and the cancellation token is set to true, then the task will not be run. If it already began , then you are correct. Agree that the Task.Run was just a convenience thing for now.

Reply:
That's true. But if that's the goal, then just calling `cancellationToken.ThrowIfCancellationRequested()` is way less expensive than spinning up another thread. Though I don't think doing that check here is a good place.